### PR TITLE
router.js do not load all entities, when Id is missing

### PR DESCRIPTION
**Pull request by pmattmann**
This case can be observed when displaying the dashboard of a camp, where all periods (not only those of this camp, all visible ones!) are loaded from `router.js` - because the `periodId` is not set.

image

### CODE REVIEW

**Review by carlobeltrame — APPROVED** [review]

File: frontend/src/router.js [hunk 1]
```diff
@@ -704,35 +704,59 @@ async function requireChecklist(to, from, next) {
 }
 
 export function campFromRoute(route) {
-  return apiStore.get().camps({ id: route.params.campId })
+  if (route.params.campId) {
```
Comment (BacLuc) [inline frontend/src/router.js]: here i would prefer a guard clause with early return.
This is whats done here.

if you want to communicate it positively (we need the campId to fetch the camp with the correct id), then
technically, the else can be omitted.
If a function does not return anything, the return value of the function call is undefined.
But you also can communicate it explicitly that this case leads to an undefined.

**Review by BacLuc — APPROVED** [review]

File: frontend/src/router.js (hunk 1 above)
Comment (pmattmann) [reply frontend/src/router.js]: What do you mean by guard clause?

```cs
export function campFromRoute(route) {
  if (!route.params.campId) {
    return undefined
  }
  return apiStore.get().camps({ id: route.params.campId })
}
```

or

```cs
export function campFromRoute(route) {
  if (route.params.campId) {
    return apiStore.get().camps({ id: route.params.campId })
  }
}
```

Comment (BacLuc) [reply frontend/src/router.js]: Variant 1 would be the guard (with early return).

Variant 2 would be just omitting the return undefined implicitly.

Comment (pmattmann) [reply frontend/src/router.js]: I can live with both variants - choose one ;)

Comment (BacLuc) [reply frontend/src/router.js]: I choose Variant 1

Comment (pmattmann) [reply frontend/src/router.js]: done

**Pull request merged**